doc/design/record_fetcher_proposal.md

```diff
@@ -0,0 +1,167 @@
+easily refined offline, with greatly reduced risk of the extraction process tainting the original data.
+
+Separately, there is a large concern surrounding tracking of data lineage to prevent (or at least reliably detect)
+accidental or intentional tampering of data. Offline processing of the raw source data makes it possible to cheaply
+iterate on extraction tooling and forward propagate the results, while also being able to compare against previous
+extractions, without the need for querying a service with questionable availability and durability.
+
+## Requirements
+
+We need a way to store raw information pertaining to logical records. Given logical records can span documents, we also
+need to store a representation (a manifest) of those relationships. Given records are capable of changing over time, we
+need to be able to track distinct snapshots over time. Additionally, we need a service which will take this information
+and persist it in a meaningful way. Analysis of this information is not in scope.
+
+### Manifests
+
+A manifest needs to be able to represent the raw materials associated with a record. The minimal identifiers for the
+record itself are:
+* a unique ID
+* which region and record type the manifest represents (e.g., "USA/FL/Bay/Court")
+* some sort of case identifier
+* the timestamp of collection
+
+A bunch of record-identifying metadata is cool, but we also need to be able to represent the documents that compose the
+record. To that end, we'll require a list of document references. Each reference will contain:
+* a content type (we can just use [MIME
+types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) here)
+* a URI for where that document's contents are durably stored
+* an optional comment expressing where this specific document came from
+
+In order for us to trust our collected record results, we need to be able to verify that we trust the software which
+performed the collection. This means being able to audit the code that created each manifest. To this end we also need
+to include in our manifest:
+* fetcher metadata
+* what's our entrypoint for this scraper?
+* what commit hash are we running from?
+* a digital signature of the rest of the manifest
+	* this is required to anchor our data lineage
+	* key management is out of scope for this document (but the appendix touches on a half-baked approach)
+* support for extension in the future (we can't be locked in to just these fields)
+
+### Storage Endpoints
+We need to be able to store the raw documents and manifests somewhere durable. This document doesn't have strong
+opinions regarding how things are stored, but we need a specific API to ensure safe collection. Our collection API:
+* must allow writing new documents and manifests
+* must not allow for the fetching or altering or existing documents or manifests
+* must enforce authentication to prevent dumping unrelated documents
```

Comment:
What is considered an unrelated document?

Reply:
Expanded upon this a bit. I was mostly interested in untrusted entities soaking up all of our money by trying to store /dev/urandom of whatever.